src/transformers/models/depth_anything/modeling_depth_anything.py

```diff
@@ -0,0 +1,471 @@
+# coding=utf-8
+# Copyright 2024 TikTok and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Depth Anything model."""
+
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+
+from ...file_utils import (
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from ...modeling_outputs import DepthEstimatorOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import logging
+from ..auto import AutoBackbone
+from .configuration_depth_anything import DepthAnythingConfig
+
+
+logger = logging.get_logger(__name__)
+
+# General docstring
+_CONFIG_FOR_DOC = "DepthAnythingConfig"
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "tiktok/depth-anything-small"
+_EXPECTED_OUTPUT_SHAPE = [1, 577, 1024]
+
+
+DEPTH_ANYTHING_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "tiktok/depth-anything-small",
+    # See all Depth Anything models at https://huggingface.co/models?filter=depth_anything
+]
+
+
+class DepthAnythingReassembleStage(nn.Module):
+    """
+    This class reassembles the hidden states of the backbone into image-like feature representations at various
+    resolutions.
+
+    This happens in 3 stages:
+    1. Take the patch embeddings and reshape them to image-like feature representations.
+    2. Project the channel dimension of the hidden states according to `config.neck_hidden_sizes`.
+    3. Resizing the spatial dimensions (height, width).
+
+    Args:
+        config (`[DepthAnythingConfig]`):
+            Model configuration class defining the model architecture.
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.config = config
+        self.layers = nn.ModuleList()
+        for i, factor in zip(range(len(config.neck_hidden_sizes)), config.reassemble_factors):
+            self.layers.append(
+                DepthAnythingReassembleLayer(config, channels=config.neck_hidden_sizes[i], factor=factor)
+            )
```

Comment:
We can just iterate over `config.neck_hidden_sizes` directly instead of indexing 

```suggestion
        for channels, factor in zip(config.neck_hidden_sizes, config.reassemble_factors):
            self.layers.append(
                DepthAnythingReassembleLayer(config, channels=channels, factor=factor)
            )
```